UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js

```diff
@@ -31,3 +31,47 @@ function RnwFilterList(pageSelectedIndex) {
     rnwUrl = rnwUrl.replace("amp;", "");
     window.location.href = rnwUrl;
 }
+
+function do_Selection() {
+    var checkboxes = document.getElementsByName('checkbox');
+    var button = document.getElementById('select_button');
+
+    if (button.value == 'Select all') {
+        for (var i in checkboxes) {
+            checkboxes[i].checked = 'FALSE';
+            document.getElementById("BtnShowSelection").disabled = false;
+        }
+        button.value = 'Clear all'
+    } else {
+        for (var i in checkboxes) {
+            checkboxes[i].checked = '';
+            document.getElementById("BtnShowSelection").disabled = true;
+        }
+        button.value = 'Select all';
+    }
+}
+
+function showSelection_() {
+    var checkboxes = document.getElementsByName('checkbox');
+    var len_ = checkboxes.length;
+    var arr = [];
+
+    for (var i in checkboxes) {
+        if (checkboxes[i].checked && i < len_) {
```

Comment:
Can directly use checkboxes.length and remove variable len_ .
```suggestion
        if (checkboxes[i].checked && i < checkboxes.length) {
```

Reply:
done required changes